File: C++/513. Find Bottom Left Tree Value.cpp
class Solution {
public:
    int findBottomLeftValue(TreeNode* root) {        
        queue<TreeNode*>q;
        q.push(root);
        int ans=0;
        while(!q.empty())
        {
            int ct=q.size();            
            for(int i=0;i<ct;i++)
            {
                TreeNode*ptr=q.front();
                q.pop();                
                if(ptr->left!=NULL)
                    q.push(ptr->left);
                if(ptr->right!=NULL)
                    q.push(ptr->right);                
                if(i==0)
                    ans=ptr->val;
            }
              
        }        
        return ans;
    }
};

File: JAVA/513. Find Bottom Left Tree Value.java
class Solution {
    int left_most_node_of_last_level = -999;
    public void breadth_first_search(TreeNode root_node){
        left_most_node_of_last_level = root_node.val;
        Queue<TreeNode> queue = new LinkedList<TreeNode>();
        queue.add(root_node);
        while(!queue.isEmpty()){
            left_most_node_of_last_level = queue.peek().val;
            for(int i=1 , N = queue.size() ; (i <= N) ; i++){
                TreeNode current_node = queue.poll();
                if(current_node.left != null){
                    queue.add(current_node.left);
                }
                if(current_node.right != null){
                    queue.add(current_node.right);
                }
            }
        }
    }
    public int findBottomLeftValue(TreeNode root_node){
        breadth_first_search(root_node);
        return left_most_node_of_last_level;
    }
}
